src/mlpack/methods/ann/loss_functions/KLDivergence.hpp

```diff
@@ -0,0 +1,112 @@
+/**
+ * @file KLDivergence.hpp
+ * @author Dakshit Agrawal
+ *
+ * Definition of the KL Divergence error function.
+ *
+ * mlpack is free software; you may redistribute it and/or modify it under the
+ * terms of the 3-clause BSD license.  You should have received a copy of the
+ * 3-clause BSD license along with mlpack.  If not, see
+ * http://www.opensource.org/licenses/BSD-3-Clause for more information.
+ */
+
+#ifndef MLPACK_METHODS_ANN_LOSS_FUNCTION_KLDIVERGENCE_HPP
+#define MLPACK_METHODS_ANN_LOSS_FUNCTION_KLDIVERGENCE_HPP
+
+#include <mlpack/prereqs.hpp>
+
+namespace mlpack {
+namespace ann /** Artificial Neural Network. */ {
+
+/**
+ * The KL divergence error function measures the network's
+ * performance according to the KL Divergence between input and
+ * target distributions.
+ *
+ * @tparam InputDataType Type of the input data (arma::colvec, arma::mat,
+ *         arma::sp_mat or arma::cube).
+ * @tparam OutputDataType Type of the output data (arma::colvec, arma::mat,
+ *         arma::sp_mat or arma::cube).
+ */
+template <
+        typename InputDataType = arma::mat,
+        typename OutputDataType = arma::mat
+>
+class KLDivergence
+{
+ public:
+  /**
+   * Create the KLDivergence object with the specified parameters.
+   *
+   * @param takeMean Boolean variable to specify whether to take mean or not.
+   */
+  KLDivergence(const bool takeMean = false);
```

Comment:
What do you think, should we just call it ``mean``?

Reply:
I think it would not specify it's real purpose. It is meant for specifying whether we have to take the mean or use the sum of individual loss.

Reply:
I see the point, thanks for the clarification.